source/Xunit.Gherkin.Quick.ProjectConsumer/Placeholders/Placeholders.cs

```diff
@@ -6,26 +6,53 @@ namespace Xunit.Gherkin.Quick.ProjectConsumer.Placeholders
     [FeatureFile("./Placeholders/Placeholders.feature")]
     public sealed class Placeholder : Feature
     {
-        private DataTable _data;
+        private DataTable _tableData;
+        private DocString _stringData;
 
         [Given("I have supplied a DataTable with")]
-        public void Given_I_Have_Supplied_DataTable(DataTable data)
+        public void I_Have_Supplied_DataTable(DataTable data)
```

Comment:
Will do, but I was following the documentation examples. I understand that they may be old and outdated in terms of semantics, but it's a bit confusing. 

[Example1](https://github.com/ttutisani/Xunit.Gherkin.Quick/blob/master/docs/docstring-argument.md), [Example2](https://github.com/ttutisani/Xunit.Gherkin.Quick/blob/master/docs/before-after-scenario-hooks.md)

Reply:
For me, personally, after writing a huge amount of steps shared between same descriptions with different adverbs (Given, Then, When) based on an already existing .feature-files codebase, it's more logical for step methods to consist of only the text description. 

This way, if you want to add another use case for this method, you won't need to rename it (still could be confusing if the text is different as well, but it's another case). And we don't need to read the method to understand what it does as it has the needed attribute right near the name. 

A popular Java's framework named Cucumber (which I already mentioned here), for example, even considers "Then", "When", "Given", etc substitutable, meaning that you can declare such method:
```
[When(@"I do something")]
void I_Do_Something() {}
```

And use it for all these lines:
```
Given I do something
When I do something
Then I do something
```

So having the adverb as a part of the method's name is not a case for some of the BDD's usages I know.

But it's just my experience and thoughts, I'm fully okay with any format you'd like your code to have!

Reply:
I authored the code, but it's not my code. It's open source 🙂

But, since you mentioned about reusing the method across multiple use-cases. I think you reuse "Given" to be another use-case's "Given" (or "And" that follows "Given"). But you won't be able to to reuse a "Given" method somewhere else as "When" or especially as "Then". These three constructs have entirely different meaning as you know - Given is preconditions (no assertions, only set-ups), When is an action you are testing (no assertions, ideally no set-ups, only invocations), and Then is assertions. Unless you are mixing the intent of these keywords in the method, you will not be able to reuse a method for another intent than what it already has.

This method, for example, is assigning a datatable to a variable. It has no assertions, it's just a set-up. So the method name explaining what's inside and how the method can be used makes most sense. This often removes the necessity to read inside the method when trying to reuse it for another use-case.

Reply:
> Will do, but I was following the documentation examples. I understand that they may be old and outdated in terms of semantics, but it's a bit confusing.
> 
> [Example1](https://github.com/ttutisani/Xunit.Gherkin.Quick/blob/master/docs/docstring-argument.md), [Example2](https://github.com/ttutisani/Xunit.Gherkin.Quick/blob/master/docs/before-after-scenario-hooks.md)

Yes, those examples are very old. so much changed since then.